lib/deploy/hosting/deploy.js

```diff
@@ -32,26 +32,41 @@ module.exports = function(context, options, payload) {
 
     return new RSVP.Promise(function(resolve, reject) {
       var lastCount = 0;
-
-      var bar = new ProgressBar(chalk.bold('Uploading:') + ' [:bar] :percent', {
-        total: upload.manifest.length,
-        width: 40,
-        complete: chalk.green('='),
-        incomplete: ' ',
-        clear: true
-      });
+      var lastPercent = 0;
+      var bar;
+      if (!options.nonInteractive || !process.stderr) {
```

Comment:
I'm not sure this logic is correct. In the case where `!process.stderr` is `true`, we don't want to use the `ProgressBar` module since it may throw (as we have seen). I think the logic we want is `if (process.stderr && !options.nonInteractive)`.